### ISSUE TITLE
"C" Language parser for FunctionList hangs forever on long list of strings

### ISSUE THREAD

**Issue by swgmike** [notepad-plus-plus/notepad-plus-plus#12497]
### Description of the Issue
Using notepad v8.4.6 and selecting the "C" Language to parse for Function Lists, notepad will hang forever i.e. 15 minutes plus on long lists of strings that are between two to 30 characters long each.
To stop it, notepad must be killed in the task manager. 

### Steps to Reproduce the Issue
1. Paste a long list of strings between two to 30 characters each into the notepad editor. In my case I'm testing my udl to colorize the keyword lists that were pasted into the editor. The UDL editor doesn't cause this problem.
2. My string lists are up to 1500 lines.  If only 150 lines are in the notepad editor, it will take about 10 seconds to show only the filename in the function list panel, and no functions. This is correct, since the list of strings don't contain functions.
3. If the list is around 1500 lines or more and the function list "C" parser is applied, notepad will hang forever. However, if any other
language parser such as fortran, CPP, C#, java etc is applied, it will not cause notepad to hang forever.

### Expected Behavior
Expected no function list data to appear in the function list panel except the filename being parsed since the 1500 lines are just strings.
In this case the list is all the content of all the keyword lists being used for my udl to test the colorization of them.

### Actual Behavior
When the "C" language function list parser is applied, notepad will hang for ever on lists longer than 150 strings whereas other language parsers will run quickly.  There may be similar issues when trying to process a line that exceeds 8000 chars (i.e. the max Unix stream in/out buffer characters and may be the same for windows).

### Debug Information
Notepad++ v8.4.6   (64-bit)
Build time : Sep 25 2022 - 19:51:39
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : "C:\Users\mike\Documents\DOORS_RPE_Library\DOORS_DXL_Examples\NotePPlus_DXL_tool\DXL.xml"
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 8.1 (64-bit) 
OS Build : 9600.20671
Current ANSI codepage : 1252
Plugins : 
    mimeTools (2.8)
    NppConverter (4.4)
    NppExport (0.4)

**Comment by benmurrell** [comment]
I was able to reproduce the hanging behavior with [invalid-c.txt](https://github.com/notepad-plus-plus/notepad-plus-plus/files/10015740/invalid-c.txt).

Steps:
1. Open the "invalid-c.txt" file
2. Select the C language by clicking "Language" on the top bar, then "C", then "C"
3. Enable the Function List view by clicking "View" on the top bar, then "Function List"

The hang can be quickly reproduced by switching to another tab then switching back to "invalid-c.txt".

Note that adding a { to a line later in the file reduces the hang time significantly. For example, adding a { on line 500 makes the hang imperceptible.

**Comment by donho** [comment]
@swgmike 
I cannot reproduce the infinite hanging.
Could you provide file to reproduce the problem?

**Comment by swgmike** [comment]
Since I've submitted this issue, I've updated to windows 10, and recently npp 8.5.3.
I've applied the "C" languages and the function list to my list of keywords 1-5 that I use in my language (since C-like) it will display an empty function list as expected. If any keyword appends (){} to the end it will display it in the function list. Since my list is long, it can take 30-40 secs to do this. This is alot better than forever.
If apply the invalid c.txt list that the person above, it will also work now too.

So based on the latest npp release, it looks like the problem was fixed. See TomPomee for bug #13664 that was posted.

So I'll close at this time as being fixed.
